client/my-sites/media-library/external-media-header.jsx

```diff
@@ -123,28 +159,148 @@ class MediaLibraryExternalHeader extends React.Component {
 		return <span className="media-library__pexels-attribution">{ attribution }</span>;
 	}
 
-	renderCard() {
-		const { onMediaScaleChange, translate, canCopy, hasRefreshButton, hasAttribution } = this.props;
+	onToggleChange( value ) {
+		return function() {
+			this.setState( {
+				mediaViewType: value,
+			} );
+
+			const folder = value === 'recent' ? 'recent' : '/';
+			this.props.onFolderChange( folder );
+		}.bind( this );
+	}
 
+	renderCard() {
 		return (
 			<Card className="media-library__header">
-				{ hasAttribution && this.renderPexelsAttribution() }
+				{ this.renderPexelsAttribution() }
+
+				{ this.renderPhotosViewToggle() }
+
+				{ this.renderBackButton() }
 
-				{ hasRefreshButton && (
-					<Button compact disabled={ this.state.fetching } onClick={ this.handleClick }>
-						<Gridicon icon="refresh" size={ 24 } />
+				{ this.renderRefreshButton() }
 
-						{ translate( 'Refresh' ) }
-					</Button>
-				) }
+				{ this.renderCopyButton() }
 
-				{ canCopy && this.renderCopyButton() }
+				{ this.renderFolderDropdown() }
 
-				<MediaLibraryScale onChange={ onMediaScaleChange } />
+				<MediaLibraryScale onChange={ this.props.onMediaScaleChange } />
 			</Card>
 		);
 	}
 
+	folderViewActive() {
+		return this.state.mediaViewType === 'folders';
+	}
+
+	showBackButton() {
+		const { hasFolders, folder } = this.props;
+
+		const folderViewActive = this.folderViewActive();
+
+		return hasFolders && folderViewActive && folder !== '/' && folder !== 'recent';
+	}
+
+	renderFolderDropdown() {
+		if (
+			! config.isEnabled( 'external-media/google-photos/folder-dropdown' ) ||
+			! this.folderViewActive()
+		) {
+			return null;
+		}
+
+		const foldersWithPhotos = this.props.folders.filter( folderItem => folderItem.children );
+
+		return (
+			<MediaFolderDropdown
+				className="media-library__header-item"
+				disabled={ this.state.fetching }
+				onFolderChange={ this.props.onFolderChange }
+				folders={ foldersWithPhotos }
+				folder={ this.props.folder }
+				defaultOption={ {
+					ID: '/',
+					name: this.props.translate( 'All Albums' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Albums' )` *ES Score: 19*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Reply:
@ehg Looks like `Albums` has already been translated. Is it best to

1. Call `translate()` twice - once for `All` and again for `Albums`?
2. Utilise [String Substitution](https://github.com/Automattic/i18n-calypso#string-substitution)?
3. Leave "as is"?

Appreciate your advice here.

Reply:
I think it's fine as is.